File: Core/YABACore/src/commonMain/kotlin/dev/subfly/yabacore/state/selection/bookmark/BookmarkSelectionUIState.kt
package dev.subfly.yabacore.state.selection.bookmark

import androidx.compose.runtime.Immutable
import dev.subfly.yabacore.model.ui.BookmarkUiModel

@Immutable
data class BookmarkSelectionUIState(
    val query: String = "",
    val bookmarks: List<BookmarkUiModel> = emptyList(),
    val selectedBookmarkId: String? = null,
    val selectedBookmark: BookmarkUiModel? = null,
    val isLoading: Boolean = false,
)

File: Darwin/YABACore/State/Selection/Bookmark/BookmarkSelectionUIState.swift
//
//  BookmarkSelectionUIState.swift
//  YABACore
//

import Foundation

public struct BookmarkSelectionUIState: Sendable {
    public var selectedBookmarkId: String?
    public var query: String

    public init(selectedBookmarkId: String? = nil, query: String = "") {
        self.selectedBookmarkId = selectedBookmarkId
        self.query = query
    }
}
